### PR TITLE
(feat) Initial work to support SHA validity check and setting SHA insuarance expiry date

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR adds functionality to simulate hitting Social Health Authority to verify that SHA Number is valid. cc @jecihjoy Will need to include `Social Health Insurance Fund (SHA)` as a method of payment to initial system setup. 

## Screenshots
https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/28008754/3e5b0adc-2865-4359-bf28-a0da59f8f1d8

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      // TODO: set validity based on api response and update the expiry date based on the response as visit attribute
+      randomNumber % 2 === 0 ? setValidity(false) : setValidity(true);
+      setMessage(
+        randomNumber % 2 === 0
+          ? t('invalidSHANumber', 'SHA number is invalid, advice patient to update payment or contact SHA')
+          : t('validSHANumber', 'SHA number is valid, proceed with care'),
+      );
+      setIsLoading(false);
+    }, 1500);
+  };
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx:31]: TODO, replace this function with API once available

**Review by jecihjoy — APPROVED** [review]

File: packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx (hunk 1 above)
Comment (ojwanganto) [reply packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx:31]: Makes sense, @donaldkibet.
The validity check, I guess, is for subscription so we can consider editing the labels to reflect subscription or something related

File: packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  paymentMethod: any;
+};
+
+const SHANumberValidity: React.FC<SHANumberValidityProps> = ({ paymentMethod }) => {
+  const { t } = useTranslation();
+  const [shaNumber, setNumber] = useState('');
+  const [message, setMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [validity, setValidity] = useState(false); // TODO: set validity based on api response
+  const isSHA = paymentMethod?.name === 'Social Health Insurance Fund (SHA)';
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx:15]: We can confirm the most appropriate name, but we can consider SHA scheme which includes SHIF, Primary Health Care Fund, and the Emergency and Chronic ... Any opinion?

File: packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  const handleValidateSHANumber = () => {
+    setIsLoading(true);
+    const randomNumber = Math.floor(Math.random() * 10) + 1;
+    // TODO call api to validate sha number
+    setTimeout(() => {
+      // TODO: set validity based on api response and update the expiry date based on the response as visit attribute
+      randomNumber % 2 === 0 ? setValidity(false) : setValidity(true);
+      setMessage(
+        randomNumber % 2 === 0
+          ? t('invalidSHANumber', 'SHA number is invalid, advice patient to update payment or contact SHA')
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx:26]: Let's use something like 'Patient has no/outdated SHA subscription'

File: packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          : t('validSHANumber', 'SHA number is valid, proceed with care'),
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/billing-form/social-health-authority/sha-number-validity.component.tsx:27]: We can use something close to 'Patient has active SHA subscription'

**Review by ojwanganto — APPROVED** [review]
We can address the comments in a followup PR. Thanks so much Don

**Pull request merged**